src/main/java/core/basesyntax/FigureSupplier.java

```diff
@@ -0,0 +1,29 @@
+package core.basesyntax;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int MAX_FIGURE_ID = 5;
+    private static final int MAX_SIZE = 50;
+    private final Random random = new Random();
+    private final ColorSupplier colorSupplier = new ColorSupplier();
+
+    public Figure getRandomFigure() {
+        return switch (random.nextInt(MAX_FIGURE_ID)) {
+            case 0 -> new Square(random.nextDouble(MAX_SIZE), colorSupplier.getRandomColor());
+            case 1 -> new Rectangle(random.nextDouble(MAX_SIZE), random.nextDouble(MAX_SIZE),
+                    colorSupplier.getRandomColor());
+            case 2 -> new RightTriangle(random.nextDouble(MAX_SIZE), random.nextDouble(MAX_SIZE),
+                    colorSupplier.getRandomColor());
+            case 3 -> new Circle(random.nextDouble(MAX_SIZE), colorSupplier.getRandomColor());
+            case 4 -> new IsoscelesTrapezoid(
+                    random.nextDouble(MAX_SIZE), random.nextDouble(MAX_SIZE),
+                    random.nextDouble(MAX_SIZE), colorSupplier.getRandomColor());
+            default -> getDefaultFigure();
+        };
+    }
+
+    public Figure getDefaultFigure() {
+        return new Circle(10, Colors.WHITE.name());

```

Comment:
You can extract default radius to constant.
If you make the `color` field immutable (make it `final`), you might as well extract the whole default `Circle` object to constant.